hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiveLog.java

```diff
@@ -491,6 +519,166 @@ public void testConvertCommitMetadata() {
+  }
+
+  @Test
+  public void testArchiveCompletedShouldNotArchiveIfInstantsLessThanMaxtoKeep() throws IOException {
+    int minInstants = 2;
+    int maxInstants = 10;
+    HoodieWriteConfig cfg =
+            HoodieWriteConfig.newBuilder().withPath(basePath).withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA)
+                    .withParallelism(2, 2).forTable("test-trip-table")
+                    .withCompactionConfig(HoodieCompactionConfig.newBuilder().retainCommits(1).archiveCommitsWith(minInstants, maxInstants).build())
+                    .build();
+    metaClient = HoodieTableMetaClient.reload(metaClient);
+    for (int i = 0; i < maxInstants; i++) {
+      createCleanMetadata(i + "", false);
+    }
+
+    HoodieTable table = HoodieSparkTable.create(cfg, context, metaClient);
+    HoodieTimelineArchiveLog archiveLog = new HoodieTimelineArchiveLog(cfg, table);
+
+    archiveLog.archiveIfRequired(context);
+    assertEquals(maxInstants, metaClient.getActiveTimeline().reload().getInstants().count());
+  }
+
+  @Test
+  public void testArchiveCompletedRollbackAndClean() throws IOException {
+    int minInstantsToKeep = 2;
+    int maxInstantsToKeep = 10;
+    HoodieWriteConfig cfg =
+            HoodieWriteConfig.newBuilder().withPath(basePath).withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA)
+                    .withParallelism(2, 2).forTable("test-trip-table")
+                    .withCompactionConfig(HoodieCompactionConfig.newBuilder().retainCommits(1).archiveCommitsWith(minInstantsToKeep, maxInstantsToKeep).build())
+                    .build();
+    metaClient = HoodieTableMetaClient.reload(metaClient);
+
+    int startInstant = 1;
+    for (int i = 0; i < maxInstantsToKeep + 1; i++, startInstant++) {
+      createCleanMetadata(startInstant + "", false);
+    }
+
+    for (int i = 0; i < maxInstantsToKeep + 1; i++, startInstant += 2) {
+      createCommitAndRollbackFile(startInstant + 1 + "", startInstant + "", false);
+    }
+
+    HoodieTable table = HoodieSparkTable.create(cfg, context, metaClient);
+    HoodieTimelineArchiveLog archiveLog = new HoodieTimelineArchiveLog(cfg, table);
+
+    archiveLog.archiveIfRequired(context);
+
+    Stream<HoodieInstant> currentInstants = metaClient.getActiveTimeline().reload().getInstants();
+    Map<Object, List<HoodieInstant>> actionInstantMap = currentInstants.collect(Collectors.groupingBy(HoodieInstant::getAction));
+
+    assertTrue(actionInstantMap.containsKey("clean"), "Clean Action key must be preset");
+    assertEquals(minInstantsToKeep, actionInstantMap.get("clean").size(), "Should have min instant");
+
+    assertTrue(actionInstantMap.containsKey("rollback"), "Rollback Action key must be preset");
+    assertEquals(minInstantsToKeep, actionInstantMap.get("rollback").size(), "Should have min instant");
+  }
+
+  @Test
+  public void testArchiveInflightRollbackAndClean() throws IOException {
```

Comment:
does this test include rollbacks? can you change name appropriately?

Reply:
Fixed